torch/fx/experimental/proxy_tensor.py

```diff
@@ -202,28 +217,43 @@ def __init__(self, tracer):
 
     def __torch_dispatch__(self, func_overload, types, args=(), kwargs=None):
         func = func_overload.overloadpacket
+        if func_overload == torch.ops.prim.device.default:
+            return args[0].device
         if any(tuple(isinstance(arg, ProxyTensor) for arg in pytree.tree_flatten(args)[0])):
             return proxy_call(func_overload, args, kwargs)
         else:
             proxy_out = self.tracer.create_proxy('call_function', func, args, kwargs,
                                                  name=self.tracer.graph._target_to_str(func.__name__))
 
-            with no_dispatch():
-                real_out = func_overload(*args, **kwargs)
+            real_out = func_overload(*args, **kwargs)
 
             return wrap_output(real_out, proxy_out)
 
 
-def make_fx(f, decomposition_table=None, trace_factory_functions=True):
+def make_fx(f, decomposition_table=None, trace_factory_functions=True, use_fake=False):
     if decomposition_table is None:
         decomposition_table = {}
 
     @functools.wraps(f)
     def wrapped(*args):
-        phs = pytree.tree_map(lambda x: fx.PH, args)  # type: ignore[attr-defined]
+        phs = pytree.tree_map(lambda _: fx.PH, args)  # type: ignore[attr-defined]
+        fx_tracer = PythonKeyTracer()
+        fake_tensor_mode = FakeTensorMode() if use_fake else nullcontext()
+        proxy_mode = ProxyTorchDispatchMode(fx_tracer) if trace_factory_functions else nullcontext()
+
+        def wrap_fake(x):
+            if isinstance(x, torch.Tensor):
+                return FakeTensor(fake_tensor_mode, x.to('meta'), x.device)
+            return x
+
         with decompose(decomposition_table):
-            t = dispatch_trace(wrap_key(f, args), concrete_args=tuple(phs),
-                               trace_factory_functions=trace_factory_functions)
+            with fake_tensor_mode:  # type: ignore[attr-defined]
+                with proxy_mode:  # type: ignore[attr-defined]
+                    if use_fake:  # type: ignore[attr-defined]
+                        with no_dispatch():
```

Comment:
Why not? Won't the proxytensor/faketensor mode intercept any constructors inside of the `FakeTensor.from_mode` otherwise?

I guess I could just move it out of the context though, but I do want to understand why you think it isn't needed?

Reply:
It will but is that an issue? That context manager should support such calls.

Anyways, moving it out also works!

Reply:
It's specifically the ProxyTensor interception that's problematic here I believe.